FaroShuffle/FaroShuffle/Extensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+
+namespace FaroShuffle
+{
+    public static class Extensions
+    {
+        public static IEnumerable<T> InterleaveSequenceWith<T>(this IEnumerable<T> first, IEnumerable<T> second)
+        {
+            var firstIter = first.GetEnumerator();
+            var secondIter = second.GetEnumerator();
+
+            while (firstIter.MoveNext()  && secondIter.MoveNext())  // note for self - enumerator starts before 1st element
+                                                                    // must call MoveNext to advance to first element before reading current
+            {
+                yield return firstIter.Current;
+                yield return secondIter.Current;
+            }
+        }
+
+        public static bool SequenceEquals<T>(this IEnumerable<T> first, IEnumerable<T> second)
+        {
+            var firstIter = first.GetEnumerator();
+            var secondIter = second.GetEnumerator();
+            
+            while (firstIter.MoveNext() && secondIter.MoveNext())
+            {
+                if (!firstIter.Current.Equals(secondIter.Current))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        public static IEnumerable<T> LogQuery<T>(this IEnumerable<T> sequence, string tag)
```

Comment:
Comments.

Also this method should abstract the log file location and allow the external methods to define this.